AABB_tree/include/CGAL/AABB_C3T3_triangle_primitive.h

```diff
@@ -0,0 +1,115 @@
+// Copyright (c) 2009 INRIA Sophia-Antipolis (France).
+// All rights reserved.
+//
+// This file is part of CGAL (www.cgal.org).
+// You can redistribute it and/or modify it under the terms of the GNU
+// General Public License as published by the Free Software Foundation,
+// either version 3 of the License, or (at your option) any later version.
+//
+// Licensees holding a valid commercial license may use this file in
+// accordance with the commercial license agreement provided with the software.
+//
+// This file is provided AS IS with NO WARRANTY OF ANY KIND, INCLUDING THE
+// WARRANTY OF DESIGN, MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE.
+//
+// $URL$
+// $Id$
+//
+//
+// Author(s)     : Stéphane Tayeb, Pierre Alliez
+//
+
+#ifndef CGAL_AABB_C3T3_TRIANGLE_PRIMITIVE_H_
+#define CGAL_AABB_C3T3_TRIANGLE_PRIMITIVE_H_
+
+#include <boost/utility/enable_if.hpp>
+#include <boost/type_traits/is_same.hpp>
+
+namespace CGAL {
+    /// \ingroup PkgAABB_tree
+    /// Primitive type that wraps a facet handle of a C3T3,
+    /// which is used as id, and allows the construction of the datum on
+    /// the fly. Since only the facet handle is stored in this primitive,
+    /// the C3T3 from which the AABB tree is built should not be
+    /// deleted while the AABB tree is in use.
+    ///
+    /// \cgalModels `AABBPrimitive`
+    /// \tparam GeomTraits must provides a \c %Point_3
+    /// type, used as \c Point, and a \c %Triangle_3 type, used as \c
+    /// Datum and constructible from three arguments of type \c
+    /// Point.
+    /// \tparam  C3T3 must be a
+    /// \c CGAL::C3T3_3 whose points have type \c Point.
+    ///
+    /// \sa `AABBPrimitive`
+    /// \sa `AABB_C3T3_segment_primitive`
+    template<typename GeomTraits, typename C3T3>
+    class AABB_C3T3_triangle_primitive
+    {
```

Comment:
It looks like that class could be templated by the triangulation type, instead of `C3T3`, and thus be more generic. That would even be a good addition to the AABB tree: build a tree from a range of facets of a triangulation.